shimmingtoolbox/dicom_to_nifti.py

```diff
@@ -1,20 +1,42 @@
 # coding: utf-8
 
-# TODO: check in unit test if dcm2bids_scaffold is installed, and also check for the required version.
-
+import os
 import subprocess
+import shutil
+import tempfile
 
 from shimmingtoolbox import __dir_config_dcm2bids__
 
 
-def dicom_to_nifti(path_dicom, path_nifti, subject_id='sub-01', path_config_dcm2bids=__dir_config_dcm2bids__):
+def dicom_to_nifti(path_dicom, path_nifti, subject_id='sub-01', path_config_dcm2bids=__dir_config_dcm2bids__,
+                   special_dicom = False):
     """ Converts dicom files into nifti files by calling dcm2bids
 
     Args:
         path_dicom (str): path to the input dicom folder
         path_nifti (str): path to the output nifti folder
 
     """
-    cmd = 'dcm2bids -d {} -p {} -o {} -l DEBUG -c {}'.format(
-        path_dicom, subject_id, path_nifti, path_config_dcm2bids)
-    subprocess.run(cmd.split(' '))
+    if special_dicom:
+        with tempfile.TemporaryDirectory(prefix='st_mag_') as tmp_mag:
```

Comment:
As @jcohenadad  mentionned, copying a temporary file is not really optimized and should be rethink so it can be avoided. Also, for big files, copying can crash the OS if the computer is not powerfull enough. So it is not recommended.